packages/pluggableWidgets/datagrid-web/src/components/ColumnSelector.tsx

```diff
@@ -19,11 +20,40 @@ export function ColumnSelector(props: ColumnSelectorProps): ReactElement {
 
     useOnClickOutside([buttonRef, optionsRef], () => setShow(false));
 
+    const onClick = useCallback(
+        (isVisible: boolean, id: string) =>
+            props.setHiddenColumns(prev => {
+                if (!isVisible) {
+                    prev.splice(
+                        prev.findIndex(v => v === id),
+                        1
+                    );
+                    return [...prev];
```

Comment:
Why always return new array? (I know u copied this, but just wondering now as well)

Reply:
Otherwise setState doesnt re-render arrays..

Reply:
But this line is fixed with the below comment